### PR TITLE
Move `getDataType` method to abstract class

### PR DESCRIPTION
**Pull request by GeoSot**
As almost all Handlers keep a naming pattern, I added an Abstraction, which provides the same name based on the existing convention

### CODE REVIEW

File: src/DataType/FloatHandler.php [hunk 1]
```diff
@@ -10,13 +10,8 @@ class FloatHandler extends ScalarHandler
...
-    protected $type = 'double';
-
-    /**
-     * {@inheritdoc}
-     */
-    public function getDataType(): string
+    protected function getType(): string
     {
-        return 'float';
+        return 'double';
```
Comment (frasmage) [inline src/DataType/FloatHandler.php:15]: would this not be a breaking change? Anything already encoded as `float` would not be unserializable anymore

File: src/DataType/NullHandler.php [hunk 2]
```diff
@@ -10,13 +10,8 @@ class NullHandler extends ScalarHandler
...
-    protected $type = 'NULL';
-
-    /**
-     * {@inheritdoc}
-     */
-    public function getDataType(): string
+    public function getType(): string
     {
-        return 'null';
+        return 'NULL';
```
Comment (frasmage) [inline src/DataType/NullHandler.php:15]: similar concerns here. Will we still pick up the right handler to unserialize with the different casing?

**Review by frasmage — CHANGES_REQUESTED** [review]
The abstraction is cute, but I feel like it doesn't provide a ton of actual value. Having to reflect each handler class every time a value is loaded is a significant amount of additional computation overhead. However, these type classes don't tend to change frequently, so I don't see how this is helping the developer in exchange for the overhead.

I'm also a little concerned about some types whose name is changing.

File: src/DataType/NullHandler.php (hunk 2 above)
Comment (GeoSot) [reply src/DataType/NullHandler.php:15]: `getType` returns the old `this->type`

File: src/DataType/FloatHandler.php (hunk 1 above)
Comment (GeoSot) [reply src/DataType/FloatHandler.php:15]: Same here as `getType` returns the 'old' `$this->type`

**Comment by GeoSot** [comment]
I left the tests same, for the only reason, that they correctly check naming and types. So no breaking changes here.

As for the naming automation, I suppose will be slightly slower as it is not reflection but just passes through some functions.

_note: the reason of  this proposal, is the future thought of handlers supporting Casters_

**Comment by frasmage** [comment]
In that case, perhaps just add a base class for the new functionality that you are looking to add as part of that change, but leave the methods as is, where possible. I don't think the inference magic is worth it here.

**Pull request closed without merging**